lib/db/mysql.js

```diff
@@ -752,6 +753,59 @@ module.exports = function (log, error) {
   }
 
 
+  // USER EMAILS
+  // Insert : emails
+  // Values : normalizedEmail = $1, email = $2, uid = $3, emailCode = $4, isVerified = $5, isPrimary = $6, verifiedAt = $7, createdAt = $8
+  var CREATE_EMAIL = 'CALL createEmail_1(?, ?, ?, ?, ?, ?, ?, ?)'
+  MySql.prototype.createEmail = function (uid, data) {
+    return this.write(
+      CREATE_EMAIL,
+      [
+        data.normalizedEmail,
+        data.email,
+        uid,
+        data.emailCode,
+        data.isVerified,
+        data.isPrimary,
+        data.verifiedAt,
+        data.createdAt
+      ]
+    )
+  }
+
+  // Select : emails
+  // Values : uid = $1
+  var ACCOUNT_EMAILS = 'CALL accountEmails_1(?)'
+  MySql.prototype.accountEmails = function (uid) {
+    return this.readOneFromFirstResult(
+      ACCOUNT_EMAILS,
+      [
+        uid
+      ]
+    )
+  }
+
+  // Delete : emails
+  // Values : uid = $1, email = $2
+  var DELETE_EMAIL = 'CALL deleteEmail_1(?, ?)'
+  MySql.prototype.deleteEmail = function (uid, email) {
+    return this.write(
+      DELETE_EMAIL,
+      [
+        uid,
+        email
+      ]
+    )
+      .catch(function(err){
+        // Signal exception is triggered when an attempt to
+        // delete a primary email.
+        if (err.errno === ER_SIGNAL_EXCEPTION) {
```

Comment:
+1, it feels like we should be checking for a specific errror number somehow, rather than just "did an exception occur?"